src/storage/block_compactor.cpp

```diff
@@ -0,0 +1,360 @@
+#include "storage/block_compactor.h"
+#include <algorithm>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "storage/index/bwtree_index.h"
+#include "storage/index/index_defs.h"
+#include "storage/sql_table.h"
+namespace terrier::storage {
+namespace {
+// for empty callback
+void NoOp(void * /* unused */) {}
+}  // namespace
+
+void BlockCompactor::ProcessCompactionQueue(transaction::TransactionManager *txn_manager) {
+  std::forward_list<RawBlock *> to_process = std::move(compaction_queue_);
```

Comment:
Oddly enough, this is O(n). Not sure why that would be the case on a linked list.

https://en.cppreference.com/w/cpp/container/forward_list/operator%3D

Reply:
I'll just use a queue then?